cmd/simulator/txs/agent.go

```diff
@@ -80,6 +86,11 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 			if err := a.worker.ConfirmTx(ctx, tx); err != nil {
 				return fmt.Errorf("failed to await transaction %d: %w", i, err)
 			}
+			confirmedCount++
+			if confirmedCount == totalTxs {
+				totalTime := time.Since(start).Seconds()
+				log.Info("Execution complete", "totalTxs", totalTxs, "totalTime", totalTime, "TPS", float64(totalTxs)/totalTime)
+			}
```

Comment:
should we add per-batch metrics, and also for issueTx?

Reply:
yah we probably should